modules/axhal/src/platform/aarch64_common/psci.rs

```diff
@@ -21,22 +46,128 @@ fn psci_hvc_call(func: u32, arg0: usize, arg1: usize, arg2: usize) -> usize {
     ret
 }
 
+fn psci_call(fn_id: usize, arg0: usize, arg1: usize, arg2: usize) -> usize {
+    match axconfig::PSCI_METHOD {
+        "smc" => arm_smccc_smc(fn_id, arg0, arg1, arg2),
+        "hvc" | _ => psci_hvc_call(fn_id, arg0, arg1, arg2),
+    }
+}
+
 /// Shutdown the whole system, including all CPUs.
 pub fn system_off() -> ! {
     info!("Shutting down...");
-    psci_hvc_call(PSCI_SYSTEM_OFF, 0, 0, 0);
+    psci_call(PSCI_0_2_FN_SYSTEM_OFF as usize, 0, 0, 0);
     warn!("It should shutdown!");
     loop {
         crate::arch::halt();
     }
 }
 
 /// Starts a secondary CPU with the given ID.
-///
 /// When the CPU is started, it will jump to the given entry and set the
 /// corresponding register to the given argument.
-pub fn cpu_on(id: usize, entry: usize, arg: usize) {
-    debug!("Starting core {}...", id);
-    assert_eq!(psci_hvc_call(PSCI_CPU_ON, id, entry, arg), 0);
-    debug!("Started core {}!", id);
+///
+/// PSCI v0.1
+/// cpu entry_point: phys_addr(secondary_entry)
+pub fn psci_cpu_on(is_fn64: bool, cpuid: usize, entry_point: usize, arg: usize) -> i32 {
```

Comment:
The `is_fn64` argument can be removed, since `PSCI_0_2_FN64_CPU_ON` is used in all cases.